core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -1889,6 +1891,9 @@ private long loadFileMetadataAndJournal(LockedInodePath inodePath,
     String ufsOwner = ufs.getOwner(ufsUri.toString());
     String ufsGroup = ufs.getGroup(ufsUri.toString());
     short ufsPermission = ufs.getMode(ufsUri.toString());
+    if (resolution.getShared()) {
+      ufsPermission |= (ufsPermission & (short) 0700) >> 6;
```

Comment:
if I am reading this correctly with given "others" the same permissions as the "owner", correct?

it might make sense to turn this into:

`ufsPermission.setOtherBits(ufsPermission.getOwnerBits())`

or something along those lines